tensorflow_probability/python/distributions/continuous_bernoulli.py

```diff
@@ -139,77 +153,60 @@ def _event_shape(self):
         return tf.TensorShape([])
 
     def _outside_unstable_region(self):
-        lams = self._lams_parameter_no_checks()
-        return lams < self._lims[0] | lams > self._lims[1]
+        probs = self._probs_parameter_no_checks()
+        return probs < self._lims[0] | probs > self._lims[1]
 
-    def _cut_lams(self):
-        lams = self._lams_parameter_no_checks()
+    def _cut_probs(self):
+        probs = self._probs_parameter_no_checks()
         return tf.where(
             self._outside_unstable_region(),
-            lams,
-            self._lims[0] * tf.ones_like(lams)
-        )
+            probs,
+            self._lims[0] * tf.ones_like(probs))
 
     def _sample_n(self, n, seed=None):
-        cut_lams = self._cut_lams()
-        new_shape = tf.concat([[n], tf.shape(cut_lams)], 0)
-        uniform = tf.random.uniform(new_shape, seed=seed, dtype=cut_lams.dtype)
-        sample = tf.where(
-            self._outside_unstable_region(),
-            (
-                tf.math.log1p(-cut_lams + uniform * (2.0 * cut_lams - 1.0))
-                - tf.math.log1p(-cut_lams)
-            )
-            / (tf.math.log(cut_lams) - tf.math.log1p(-cut_lams)),
-            uniform
-        )
+        cut_probs = self._cut_probs()
+        new_shape = tf.concat([[n], tf.shape(cut_probs)], axis=0)
+        uniform = tf.random.uniform(new_shape, seed=seed, dtype=cut_probs.dtype)
+        sample = self._quantile(uniform)
         return tf.cast(sample, self.dtype)
 
     def _cont_bern_log_norm(self):
-        lams = self._lams_parameter_no_checks()
-        cut_lams = self._cut_lams()
+        probs = self._probs_parameter_no_checks()
+        cut_probs = self._cut_probs()
         log_norm = tf.math.log(
-            tf.math.abs(tf.math.log1p(-cut_lams) - tf.math.log(cut_lams))
-        ) - tf.math.log(tf.math.abs(1 - 2.0 * cut_lams))
-        taylor = (
-            tf.math.log(2.0)
-            + 4.0 / 3.0 * tf.math.pow(lams - 0.5, 2)
-            + 104.0 / 45.0 * tf.math.pow(lams - 0.5, 4)
-        )
+            tf.math.abs(tf.math.log1p(-cut_probs) - tf.math.log(cut_probs))
+        ) - tf.math.log(tf.math.abs(1 - 2.0 * cut_probs))
```

Comment:
For numeric reasons, it might be worth rewriting `tf.math.log(tf.math.abs(1. - 2. * cut_probs))` as 
```
cut_probs_below_half = tf.where(cut_probs < 0.5, cut_probs, tf.zeros_like(cut_probs))
cut_probs_above_half = tf.where(cut_probs > 0.5, cut_probs, tf.ones_like(cut_probs))
tf.where(cut_probs < 0.5, tf.math.log1p(-2. * cut_probs_below_half), tf.math.log(2. * cut_probs_above_half - 1.))`
```

The reason is that at least for small `cut_probs`, we'll be able to retain some accuracy. I unfortunately have to add those double `tf.where` to ensure that gradients exist (see https://github.com/tensorflow/probability/blob/master/discussion/where-nan.pdf as an example).